src/lib.rs

```diff
@@ -469,10 +474,23 @@ impl<'b, T: Write + 'b> Session<'b, T> {
     }
 }
 
+pub(crate) fn create_emitter<'a>(config: &Config) -> Box<dyn Emitter + 'a> {
+    match config.emit_mode() {
+        EmitMode::Files if config.make_backup() => Box::new(emitter::FilesWithBackupEmitter::new()),
```

Comment:
I prefer to use `default()` over an empty constructor.

Reply:
Addressed in dc1a440